### ISSUE TITLE
UI text selection with mouse selects wrong range

### ISSUE THREAD

**Issue by evilworm** [HeidiSQL/HeidiSQL#1275]
**Describe the bug**
When working with text fields, double clicking a text in quotes selects everything from previous space character to the next space character. This is particularly annoying when working with json fields and trying to quickly modify variables.

**To Reproduce**
1. Create a varchar / text / json or any other textual field
2. insert a text like ["key": "value"]
3. try double clicking "key"
4. the selection will contain ["key":

**Expected behavior**
When double clicking a word in (single/double) quotes, I expect to select that particular word only. In fact the selection should have multiple stop-symbols (like brackets, quotes, dashes, etc). Or the selection should just do a regex [a-zA-Z0-9] and skip any character altogether. 

**Screenshots**
![heidi1](https://user-images.githubusercontent.com/5744459/103515389-4f184f00-4e77-11eb-99ad-b8cb9b2a3cfc.jpg)
![heidi2](https://user-images.githubusercontent.com/5744459/103515418-5d666b00-4e77-11eb-83c3-8c958ca9b716.jpg)

**Environment:**
 - OS: Windows 10
 - HeidiSQL 11.1.0.6116

**Comment by WesleyM77** [comment]
Might be an upstream issue with SynEdit. Ansgar will know for sure.

**Comment by ansgarbecker** [comment]
That text editor is a normal `TMemo`, with no ability to finetune the behaviour of double-clicks, apart from writing an event trigger. Not sure if it's worth to do that, especially as there is a plan to use SynEdit here in the near future. See #136
Once that is done, the text editor will select a word only on double-click.

**Comment by ansgarbecker** [comment]
Latest build  now has highlighters and a better working SynMemo, which fixes this issue:
![grafik](https://user-images.githubusercontent.com/7986591/111369677-a452b700-8697-11eb-8f33-93f9717c8b60.png)

You could update to the latest build and try that out, so we can perhaps mark this as fixed.

**Comment by evilworm** [comment]
I see that the issue is now fixed in the popup editor however if you do an inline edit of a row, the text selection still is wrong

**Comment by ansgarbecker** [comment]
Indeed, I see this is the case for all edit boxes in HeidiSQL, not just in the grid cells:
![grafik](https://user-images.githubusercontent.com/7986591/111435349-f845b600-8700-11eb-82bb-73a2959ff2df.png)

And I highly guess the current implementation is defined in the Windows API, as my Explorer does the same when renaming a file:
![grafik](https://user-images.githubusercontent.com/7986591/111435497-275c2780-8701-11eb-819d-8c131168ffc6.png)

I'm not sure if it's worth it, but this will need a replacement for the underlying edit component, with some custom code for the double-click event.

**Comment by ansgarbecker** [comment]
Related, probabe solution: https://forum.lazarus.freepascal.org/index.php?topic=18063.0
